shell/platform/embedder/tests/embedder_unittests.cc

```diff
@@ -99,5 +105,102 @@ TEST_F(EmbedderTest, CanInvokeCustomEntrypointMacro) {
+    forwarding_target_ = std::move(runner);
+  }
+
+  FlutterTaskRunnerDescription GetEmbedderDescription() {
+    FlutterTaskRunnerDescription desc;
+    desc.struct_size = sizeof(desc);
+    desc.user_data = this;
+    desc.runs_task_on_current_thread_callback = [](void* user_data) -> bool {
+      return reinterpret_cast<EmbedderTestTaskRunner*>(user_data)
+          ->forwarding_target_->RunsTasksOnCurrentThread();
+    };
+    desc.post_task_callback = [](FlutterTask task, uint64_t target_time_nanos,
+                                 void* user_data) -> void {
+      auto runner = reinterpret_cast<EmbedderTestTaskRunner*>(user_data);
+
+      auto target_time = fml::TimePoint::FromEpochDelta(
+          fml::TimeDelta::FromNanoseconds(target_time_nanos));
+
+      runner->forwarding_target_->PostTaskForTime(
+          [task, forwarder = runner->on_forward_task_]() { forwarder(task); },
+          target_time);
+    };
+    return desc;
+  }
+
+ private:
+  fml::RefPtr<fml::TaskRunner> forwarding_target_;
+  std::function<void(FlutterTask)> on_forward_task_;
+
+  FML_DISALLOW_COPY_AND_ASSIGN(EmbedderTestTaskRunner);
+};
+
+TEST_F(EmbedderTest, CanSpecifyCustomTaskRunner) {
+  auto& context = GetEmbedderContext();
+  fml::AutoResetWaitableEvent latch;
+
+  // Run the test on its own thread with a message loop so that it san safely
+  // pump its event loop while we wait for all the conditions to be checked.
+  fml::Thread thread;
+  UniqueEngine engine;
+  bool signalled = false;
+
+  EmbedderTestTaskRunner runner([&](FlutterTask task) {
+    // There may be multiple tasks posted but we only need to check assertions
+    // once.
+    if (signalled) {
+      // Since we have the baton, return it back to the engine. We don't care
+      // about the return value because the engine could be shutting down an it
+      // may not actually be able to accept the same.
+      FlutterEngineRunTask(engine.get(), &task);
+      return;
+    }
+
+    signalled = true;
+    FML_LOG(INFO) << "Checking assertions.";
+    ASSERT_TRUE(engine.is_valid());
+    ASSERT_EQ(FlutterEngineRunTask(engine.get(), &task), kSuccess);
+    latch.Signal();
+  });
+
+  thread.GetTaskRunner()->PostTask([&]() {
+    EmbedderConfigBuilder builder(context);
+    builder.AddCommandLineArgument("--verbose-logging");
+    const auto task_runner_description = runner.GetEmbedderDescription();
+    runner.SetForwardingTaskRunner(
+        fml::MessageLoop::GetCurrent().GetTaskRunner());
+    builder.SetPlatformTaskRunner(&task_runner_description);
+    builder.SetDartEntrypoint("invokePlatformTaskRunner");
+    engine = builder.LaunchEngine();
+    ASSERT_TRUE(engine.is_valid());
+  });
+
+  // Signalled when all the assertions are checked.
+  latch.Wait();
+  FML_LOG(INFO) << "Assertions checked. Killing engine.";
+  ASSERT_TRUE(engine.is_valid());
+
+  // Since the engine was started on its own thread. It must be killed there as
```

Comment:
nit: "Since ..., it must be killed there ..."

Reply:
Done